common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java

```diff
@@ -1392,35 +1440,37 @@
-            SIGN_OUT_FROM_SHARED_DEVICE(MSAL_SIGN_OUT_FROM_SHARED_DEVICE_PATH, MSAL_SIGN_OUT_FROM_SHARED_DEVICE_CODE),
-            GENERATE_SHR(GENERATE_SHR_PATH, MSAL_GENERATE_SHR_CODE),
-            BROKER_HELLO(BROKER_API_HELLO_PATH, BROKER_API_HELLO_URI_CODE),
-            BROKER_GET_ACCOUNTS(BROKER_API_GET_BROKER_ACCOUNTS_PATH, BROKER_API_GET_BROKER_ACCOUNTS_CODE),
-            BROKER_REMOVE_ACCOUNT(BROKER_API_REMOVE_BROKER_ACCOUNT_PATH, BROKER_API_REMOVE_BROKER_ACCOUNT_CODE),
-            BROKER_UPDATE_BRT(BROKER_API_UPDATE_BRT_PATH, BROKER_API_UPDATE_BRT_CODE),
-            BROKER_ADD_FLIGHTS(BROKER_API_ADD_FLIGHTS_PATH),
-            BROKER_SET_FLIGHTS(BROKER_API_SET_FLIGHTS_PATH),
-            BROKER_GET_FLIGHTS(BROKER_API_GET_FLIGHTS_PATH),
-            GET_SSO_TOKEN(GET_SSO_TOKEN_PATH),
-            UNKNOWN(null)
+            MSAL_HELLO(MSAL_HELLO_PATH, MSAL_HELLO_URI_CODE, null, "5.0"),
+            ACQUIRE_TOKEN_INTERACTIVE(MSAL_ACQUIRE_TOKEN_INTERACTIVE_PATH, MSAL_ACQUIRE_TOKEN_INTERACTIVE_CODE, null, "5.0"),
+            ACQUIRE_TOKEN_SILENT(MSAL_ACQUIRE_TOKEN_SILENT_PATH, MSAL_ACQUIRE_TOKEN_SILENT_CODE, null, "5.0"),
+            GET_ACCOUNTS(MSAL_GET_ACCOUNTS_PATH, MSAL_GET_ACCOUNTS_CODE, null, "5.0"),
+            REMOVE_ACCOUNTS(MSAL_REMOVE_ACCOUNTS_PATH, MSAL_REMOVE_ACCOUNTS_CODE, null, "5.0"),
+            GET_CURRENT_ACCOUNT_SHARED_DEVICE(MSAL_GET_CURRENT_ACCOUNT_SHARED_DEVICE_PATH, MSAL_GET_CURRENT_ACCOUNT_SHARED_DEVICE_CODE, null, "5.0"),
+            GET_DEVICE_MODE(MSAL_GET_DEVICE_MODE_PATH, MSAL_GET_DEVICE_MODE_CODE, null, "5.0"),
+            SIGN_OUT_FROM_SHARED_DEVICE(MSAL_SIGN_OUT_FROM_SHARED_DEVICE_PATH, MSAL_SIGN_OUT_FROM_SHARED_DEVICE_CODE, null, "5.0"),
+            GENERATE_SHR(GENERATE_SHR_PATH, MSAL_GENERATE_SHR_CODE, null, "6.0"),
+            BROKER_HELLO(BROKER_API_HELLO_PATH, BROKER_API_HELLO_URI_CODE, "1.0", "5.0"),
+            BROKER_GET_ACCOUNTS(BROKER_API_GET_BROKER_ACCOUNTS_PATH, BROKER_API_GET_BROKER_ACCOUNTS_CODE, "1.0", "5.0"),
+            BROKER_REMOVE_ACCOUNT(BROKER_API_REMOVE_BROKER_ACCOUNT_PATH, BROKER_API_REMOVE_BROKER_ACCOUNT_CODE, "1.0", "5.0"),
+            BROKER_UPDATE_BRT(BROKER_API_UPDATE_BRT_PATH, BROKER_API_UPDATE_BRT_CODE, "1.0", "5.0"),
+            BROKER_ADD_FLIGHTS(BROKER_API_ADD_FLIGHTS_PATH, null, "2.0", "5.0"),
+            BROKER_SET_FLIGHTS(BROKER_API_SET_FLIGHTS_PATH, null, "2.0", "5.0"),
+            BROKER_GET_FLIGHTS(BROKER_API_GET_FLIGHTS_PATH, null, "2.0", "5.0"),
+            GET_SSO_TOKEN(GET_SSO_TOKEN_PATH, null, null, "7.0"),
+            UNKNOWN(null, null, null, null)
                 ;
             private String mPath;
             private Integer mCode;
-            API(String path, int code) {
+            private String mBrokerVersion;
+            private String mMsalVersion;
+            private API(String path, Integer code, String brokerApiVersion, String msalApiVersion) {
```

Comment:
an alternate way to write this would be to have a parameter indicating whether or not the api was broker or small, and then a single version string.